src/cpp/DataBrokerConfiguration.cpp

```diff
@@ -254,6 +260,37 @@ bool DataBrokerConfiguration::load_configuration_file(
             logInfo(DATABROKER_CONFIGURATION, "Using internal domain " << configuration.local_configuration.domain);
         }
 
+        // LOCAL DISCOVERY
+        if (config_node["local-discovery"])
+        {
+            if (config_node["local-discovery"]["kind"])
+            {
+                if (config_node["local-discovery"]["kind"].as<std::string>() == "discovery-server")
+                {
+                    configuration.local_configuration.discovery_config.discovery_protocol =
+                            eprosima::fastrtps::rtps::DiscoveryProtocol::SERVER;
+
+                    configuration.local_configuration.discovery_config.listening_addresses.clear();
+                    for (auto address : config_node["local-discovery"]["listening-addresses"])
+                    {
+                        Address new_address;
+                        if (address["ip"])
+                        {
+                            new_address.ip = address["ip"].as<std::string>();
+                        }
+                        if (address["port"])
+                        {
+                            new_address.port = address["port"].as<uint32_t>();
+                        }
+                        configuration.local_configuration.discovery_config.listening_addresses.push_back(new_address);
+                    }
+
+                    configuration.local_configuration.discovery_config.server_guid =
+                            Address::ros_discovery_server_guid();
+                }
+            }
```

Comment:
If guid will not be set here, do not store it in the configuration and use always the default one. I would suggest to give the option to set it here and use `ROS_DISCOVERY_SERVER_GUID` as default (to make it more configurable)(use WAN configuration to copy `listening-addresses` configuration parser)

Reply:
Not for the PoC version